snooty/parser.py

```diff
@@ -854,6 +862,180 @@ def handle_method_option(self, node: n.Directive) -> None:
+                    )
+                    if not value_key or value_key == "None":
+                        continue
+                    option_from_spec = next(
+                        (
+                            spec_option
+                            for spec_option in composable_from_spec.options
+                            if spec_option.id == value_key
+                        ),
+                        None,
+                    )
+                    composable_option = next(
+                        (
+                            composable_option
+                            for composable_option in composable_options
+                            if composable_option["value"] == option_key
+                        ),
+                        {},
+                    )
+                    if not option_from_spec or not composable_option:
+                        continue
+                    composable_option["selections"] = (
+                        composable_option["selections"] or []
+                    )
+                    if isinstance(composable_option["selections"], list):
+                        composable_option["selections"].append(
+                            {
+                                "value": option_from_spec.id,
+                                "text": option_from_spec.title,
+                            }
+                        )
+
+                default_values_found = default_values_found or all(
+                    child.selections[composable_id] == option_id
+                    for composable_id, option_id in (default_ids_dict.items())
+                )
+
+            except ChildValidationError:
+                continue
+
+        if not default_values_found:
+            self.diagnostics.append(
+                MissingChild(
+                    "composable-tutorial",
+                    "selected-content with selections {default_ids_as_string}",
+                    node.start[0],
+                )
+            )
+        node.composable_options = composable_options
+        node.options = {}
+
+    def handle_composable_content(
+        self,
+        node: n.ComposableContent,
+        spec_composables: List[Composable],
+    ) -> None:
+        selection_ids = re.split(r"\s*,\s*", node.options.get("selections", "").strip())
+        selections: Dict[str, str] = {}
+        # validate all selection ids
+        for idx in range(len(selection_ids)):
+            selection_id = selection_ids[idx]
+            spec_composable = spec_composables[idx]
+            allowed_selection_ids = list(map(lambda x: x.id, spec_composable.options))
+            # check if dependencies are met - then None is not allowed
+            met_dependencies: bool = all(
+                selections[composable_option_id] == option_value_id
+                for composable_option_id, option_value_id in (
+                    spec_composable.dependencies or {}
+                ).items()
+            )
```

Comment:
(Question) For my own understanding, why is "None" not allowed in this case?

Reply:
All selections should have a choice. None is reserved for when that configurable option does not have its dependencies met.

example:
```
.. composable-tutorial::
   :options: interface, language, cluster-topology, cloud-provider
   :defaults: driver, nodejs, repl, gcp

   .. selected-content::
      :selections: driver, None, repl, aws
```

should result in an error because the `language` field's dependencies has been met

Reply:
Gotcha, this is a good example. Thanks!

Reply:
adding it to the tests!

Reply:
added it [here](https://github.com/mongodb/snooty-parser/pull/652/files#diff-e43a5722b09dcc8e1ea71a8a0104599496fba59b067d05ae7a7f1cb5028b6534R4698-R4707) (last case of `test_composable_tutorial_errors` )